src/libtelio.udl

```diff
@@ -437,29 +443,29 @@ dictionary FeatureNurse {
     /// The unique identifier of the device, used for meshnet ID
     string fingerprint;
     /// Heartbeat interval in seconds. Default value is 3600.
-    u32? heartbeat_interval;
+    duration heartbeat_interval;
     /// Initial heartbeat interval in seconds. Default value is None.
-    u32? initial_heartbeat_interval;
+    duration initial_heartbeat_interval;
     /// QoS configuration for Nurse
     FeatureQoS? qos;
     /// Enable/disable collecting nat type
-    boolean? enable_nat_type_collection;
+    boolean enable_nat_type_collection;
     /// Enable/disable Relay connection data
-    boolean? enable_relay_conn_data;
+    boolean enable_relay_conn_data;
     /// Enable/disable NAT-traversal connections data
-    boolean? enable_nat_traversal_conn_data;
+    boolean enable_nat_traversal_conn_data;
 };
 
 /// QoS configuration options
 dictionary FeatureQoS {
     /// How often to collect rtt data in seconds. Default value is 300.
-    u32? rtt_interval;
+    duration rtt_interval;
     /// Number of tries for each node. Default value is 3.
-    u32? rtt_tries;
+    u32 rtt_tries;
     /// Types of rtt analytics. Default is Ping.
-    sequence<string>? rtt_types;
+    sequence<RttType> rtt_types;
     /// Number of buckets used for rtt and throughput. Default value is 5.
-    u32? buckets;
+    u32 buckets;
```

Comment:
These fields without `?` mean we do not provide defaults and force the clients to set up all files properly. While your approach with removing `Option` is fine for JSON features, it doesn't port to uniffy sadly

Reply:
That is true. I was thinking about it. But I feel like for uniffi we need to rethink/redo this part anyway. First of all the functionality is not there currently. We do not have machinery to apply defaults to the config passed as struct. We do it only via serde deserialization. Secondly even if we decide to implement it I see some problems. Few examples:
1. Let's take a look at QoS. It is really optional, meaning it can be disabled. It is now set by default after this PR. When configured via JSON it can be disabled by "qos = null". It cannot be disabled via normal config because if not set we will set it by default :) So it requires some extra work. Either an extra flag or some 3 state enum like: Feature{Some, Default, Disabled};
2. It is annoying for the fields in the config to be optional it they are really not. It causes problems when you use them in the code later. You can either unwrap() and hope that they are correctly set by the defaults. Or use unwrap_or which sucks because you then need to provide a default in a second place. Possible solutions: a) do not set defaults in the config at all. Let all the users of it set it accordingly. b) Have FeaturesInput with Options and then pass it via some validator which outputs Features with Options only in places which are really optional and defaults set in other places.

So that's the reason why I've decided to not care about this part, since it is not working anyway. Let me know what are your thoughts. IMO we should add some new task to uniffi epic to refactor this part.

Reply:
Yeah, I agree with you fully. Just please ensure people are aware of that and the ticket is there

Reply:
I'm adding this convo as part of LLT-5022